### PR TITLE
[DSNP 1.2.0] Breaking Change: Use multiformat hashes

### LINKED ISSUE #19: Update for DSNP v1.2.0
The DSNP Spec is preparing to release v1.2.0 with a few minor changes to the Activity Content portion of the Spec.

- [ ] #22 Update `Hash` Data structure https://github.com/LibertyDSNP/spec/issues/210
- [x] #21 Require `published` field https://github.com/LibertyDSNP/spec/issues/158

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
DSNP 1.2.0 updated to use multiformat hashes instead of the pre-1.2.0 object format.

Closes: #19 

Solution
========
- Removed the old way completely as it was in limited use.
- Updated to the new setup for validating and creating
- Added some additional utilities around hashing.

Double Checks:
---------------
- [x] Did you update the changelog?
- [x] Any new modules need to be exported?
- [x] Are new methods in the right module?
- [x] Do you have good documentation on exported methods?

Change summary:
---------------
* Lots of tests to update
* Added the new multiformat dependencies
* Removed js-sha3 dependency
* Updated how jest runs tests some to support TextEncoder
* Now compiles the esm version instead of the commonjs version as most things support both now.

Steps to Verify:
----------------
This is not so easy, as there are not many sites out there to build multiformat hashes. I suggest looking closely at the tests, and perhaps creating some against it yourself.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,72 +1,138 @@
 {
```
Comment (wilwade) [inline .eslintrc:1]: This is mostly a reformat and some new overrides for test files

File: jest.config.ts [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-const config: Config.InitialOptions = {
-  verbose: true,
-};
 // This prevents jest from running compiled js
 // apparently it's not enough to set test root as <rootDir>/src
 module.exports = {
     moduleFileExtensions: ["ts", "js", "json"],
     moduleDirectories: ["node_modules", "<rootDir>"],
-    preset: "ts-jest",
+    preset: "ts-jest/presets/default-esm",
```
Comment (wilwade) [inline jest.config.ts:6]: Better preset

File: jest.config.ts [hunk 3]
```diff
@@ -1,22 +1,13 @@
-// jest.config.ts
-import type { Config } from "@jest/types";
-
-// Sync object
-const config: Config.InitialOptions = {
-  verbose: true,
-};
```
Comment (wilwade) [inline jest.config.ts:7]: Never did anything...? Not sure why I had it there.

File: package.json [hunk 4]
```diff
@@ -23,15 +23,16 @@
     "doc:markdown": "typedoc --options ./typedoc.json --plugin typedoc-plugin-markdown --out dist/docs",
     "format": "eslint \"**/*.{js,ts}\" --fix",
     "lint": "eslint \"**/*.{js,ts}\"",
-    "test": "jest --config ./jest.config.ts"
+    "test": "NODE_OPTIONS=\"--experimental-vm-modules --no-warnings\" jest --config ./jest.config.ts"
```
Comment (wilwade) [inline package.json:26]: Various reasons required the move to esm. This is the oddest bit there.

File: .npmignore
Comment (wilwade) [inline .npmignore:1]: Building manually, so no need for this

File: scripts/package.cjs
Comment (wilwade) [inline scripts/package.cjs:1]: Building the package.json is so much better this way I think.

File: src/hash.ts
Comment (wilwade) [inline src/hash.ts:1]: Lots of new code here! This is the heart of the changes

File: src/test/hash.test.ts
Comment (wilwade) [inline src/test/hash.test.ts:1]: And here is the new tests over the new code

File: src/validation.ts [hunk 5]
```diff
@@ -121,19 +120,23 @@ const hasAtLeastOneSupportedVideoMediaType = (
...
-): boolean => {
-  for (const hash of obj) {
-    if (
-      hash["algorithm"] === "keccak256" &&
-      isString(hash["value"]) &&
-      hash["value"].match(HASH_REGEX)
-    )
-      return true;
-  }
+const requireIsActivityContentHashArray = (
```
Comment (wilwade) [inline src/validation.ts:123]: Alert: New code! This handles the validation of hashes

File: scripts/package.cjs
Comment (JoeCap08055) [reply scripts/package.cjs:1]: Pretty cool.

**Review by JoeCap08055 — APPROVED** [review]

**Review by shannonwells — APPROVED** [review]
LGTM

**Pull request merged**